beacon-chain/rpc/prysm/v1alpha1/validator/aggregator.go

```diff
@@ -66,15 +68,22 @@ func (vs *Server) SubmitAggregateSelectionProof(ctx context.Context, req *ethpb.
 		return nil, status.Errorf(codes.InvalidArgument, "Validator is not an aggregator")
 	}
 
-	atts := vs.AttPool.AggregatedAttestationsBySlotIndex(ctx, req.Slot, req.CommitteeIndex)
-	// Filter out the best aggregated attestation (ie. the one with the most aggregated bits).
-	if len(atts) == 0 {
-		atts = vs.AttPool.UnaggregatedAttestationsBySlotIndex(ctx, req.Slot, req.CommitteeIndex)
+	var atts []*ethpb.Attestation
+
+	if features.Get().EnableExperimentalAttestationPool {
+		atts = cache.GetBySlotAndCommitteeIndex[*ethpb.Attestation](vs.AttestationCache, req.Slot, req.CommitteeIndex)
+	} else {
+		atts = vs.AttPool.AggregatedAttestationsBySlotIndex(ctx, req.Slot, req.CommitteeIndex)
+		// Filter out the best aggregated attestation (ie. the one with the most aggregated bits).
 		if len(atts) == 0 {
-			return nil, status.Errorf(codes.NotFound, "Could not find attestation for slot and committee in pool")
+			atts = vs.AttPool.UnaggregatedAttestationsBySlotIndex(ctx, req.Slot, req.CommitteeIndex)
```

Comment:
Do we want to aggregate them in this case? problem is if this returns an empty list too, we would need to add another check for length and return an error here. I kinda think it's better to just not do this check here and keep the previous behavior

Reply:
The behavior is the same, you must be reading something incorrectly

Reply:
Indeed I was!